qcodes/instrument/parameter.py

```diff
@@ -402,6 +404,17 @@ def set_wrapper(value, **kwargs):
                                              not(step_index == len(steps)-1 or
                                                  len(steps) == 1)))
 
+                    if self._snapshot_value and self.log.isEnabledFor(logging.DEBUG):
+                        # Add to log
+                        log_msg = 'parameter set to {}'.format(val_step)
+                        if mapped_value != val_step:
```

Comment:
As Travis taught us, this will fail for multi-valued parameters.

Reply:
@sohailc Perhaps you, as the author of the AMI430 driver, would like to have a look at this? What do you think, is this the fault of the driver or the logging added?

Reply:
@WilliamHPNielsen Sounds like the issue arises when comparing numpy arrays, as these cannot be compared with one another. A solution would be to use `numpy.array_equal`, although this fails when values are not the same to within machine precision. A better solution is probably `numpy.is_close`, as this lets you specify relative tolerances. In any case, I think we'll have to perform a check for numpy arrays

Reply:
@nulinspiratie that is correct but IMHO parameters that are arrays or other non scalar types should be a subclass of array or multiparameter and not hit this code path at all

Reply:
@jenshnielsen, do you then think we should rewrite the AMI430 driver? @sohailc, what's your opinion?

And if so, is there any way that we can enforce that a `Parameter` is never used to hold more than one value. Like checking for `__iter__` or `__getitem__` or what do I know?

Reply:
@jenshnielsen Do you mean they should not have the same get/set wrapper? I think we can get away with only doing a check that excludes Array and MultiParameter, either `isinstace(self, Parameter)`, or checking for attribute `shape/shapes` should do the trick

Reply:
@nulinspiratie I think that sounds sane, as we will need to handle settable multiparameters for the magnet controls. Although not directly related to this issue, we could perhaps also change the `step` setter to fail if being called from another class than `Parameter`. And also think about how to avoid ending up with `Parameters` that are not scalar.